src/features/applications/hooks/useApplicationReview.ts

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { useCallback, useEffect, useState } from "react";
+import { toast } from "sonner";
+import { useCurrentUser } from "~/hooks/useCurrentUser";
+import { useRevokeAttestations } from "~/hooks/useRevokeAttestations";
+import { useIsCorrectNetwork } from "~/hooks/useIsCorrectNetwork";
+import { useApplicationStatus } from "./useApplicationStatus";
+import { useApproveApplication } from "./useApproveApplication";
+
+const DEFAULT_REFETCH_INTERVAL = 1000 * 30;
+
+export const useApplicationReview = ({
+  projectId,
+  refetchInterval = DEFAULT_REFETCH_INTERVAL,
+}: {
+  projectId: string;
+  refetchInterval?: number;
+}) => {
+  const [isRevoking, setIsRevoking] = useState(false);
+  const [isApproving, setIsApproving] = useState(false);
+
+  const { address, isAdmin } = useCurrentUser();
+
+  const { isCorrectNetwork, switchToCorrectChain } = useIsCorrectNetwork();
+
+  const { data, refetch: refetchAttestations } = useApplicationStatus({
+    projectId,
+    withAttestations: true,
+    opts: { enabled: true, refetchInterval },
+  });
+
+  const { status, attestations } = data ?? {};
+
+  useEffect(() => {
+    switch (status) {
+      case "approved":
+        setIsApproving(false);
+        break;
+      case "rejected":
+        setIsRevoking(false);
+        break;
+    }
+  }, [status]);
```

Comment:
Let's try to not use useState or useEffect like this. Use ReactQuery useQuery instead where we have the `isPending` prop.